app/javascript/components/TestData.js

```diff
@@ -0,0 +1,23 @@
+import {gql} from "apollo-boost";
+import React from "react";
+import {useQuery} from "@apollo/react-hooks";
+
+const TEST_QUERY = gql`query { testField }`;
+
+export default function TestData() {
+    const {loading, error, data} = useQuery(TEST_QUERY);
```

Comment:
This is definitely the better way to make a query!

Reply:
Agreed! I also like the fact that this has been moved into its own file. But it's interesting since I thought we tried doing this before and it wasn't working for us. But maybe that's because we were trying to call it as a function and not returning it as a component.